### ISSUE TITLE
The training loss function drops, but the test set result also drops

### ISSUE THREAD

**Issue by wnynotfaker** [facebookresearch/detectron2#2046]
Hello, thank you very much for your work on this framework. I have some doubts during the use, I hope you can take the time to help me.

I used my own data set for training. 

First, I used a rotating bounding box for training, and then I modified the model. After I changed the model, the loss value during training decreased relative to the unchanged model. I used rotated_coco_evaluation.py to evaluate the test set. However, even if the modified model has a smaller loss value during training, the test result of the test set is worse.

In order to verify whether it is the cause of overfitting, I also tested the training set, and the result is still that the evaluation result of the modified model is lower.

It may be a problem with my model, but logically speaking, a lower loss value should indicate a better evaluation result. Detectron2 has been used and tested by so many people, and there should be no problem. But I really don't know what is the reason for this problem. If you have relevant experience, I hope you can answer it.

**Comment by ppwwyyxx** [comment]
As the [issue template](https://github.com/facebookresearch/detectron2/blob/master/.github/ISSUE_TEMPLATE/unexpected-problems-bugs.md) mentions:

> If you expect the model to converge / work better, note that we do not give suggestions on how to train a new model. Only in one of the two conditions we will help with it: (1) You're unable to reproduce the results in detectron2 model zoo. (2) It indicates a detectron2 bug.

We provide configs & models with standard academic settings and expect users to have the knowledge to choose or design appropriate models & parameters for their own tasks.
